File: frontend/src/app/services/article.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Article {
  id: number;
  nom: string;  // ✅ Assure-toi que le backend utilise "nom"
  prix: number; // ✅ Assure-toi que le backend utilise "prix"
}

@Injectable({
  providedIn: 'root'
})
export class ArticleService {
  private apiUrl = 'http://localhost:8080/api/articles'; // 🔥 Mets l'URL de ton backend Spring Boot

  constructor(private http: HttpClient) {}

  // ✅ Récupérer tous les articles
  getAllArticles(): Observable<Article[]> {
    return this.http.get<Article[]>(this.apiUrl);
  }

  // ✅ Rechercher un article par son nom exact
  getArticleByNom(nom: string): Observable<Article> {
    return this.http.get<Article>(`${this.apiUrl}/${nom}`);
  }

  // ✅ Rechercher des articles contenant un mot-clé (insensible à la casse)
  searchArticles(query: string): Observable<Article[]> {
    return this.http.get<Article[]>(`${this.apiUrl}/search?query=${query}`);
  }
}

File: backend/src/main/java/com/example/backend/service/ArticleService.java
package com.example.backend.service;

import com.example.backend.model.Article;
import com.example.backend.repository.ArticleRepository;
import org.springframework.stereotype.Service;
import java.util.Optional;
import java.util.List;

@Service
public class ArticleService {
    
    private final ArticleRepository articleRepository;

    public ArticleService(ArticleRepository articleRepository) {
        this.articleRepository = articleRepository;
    }

    // ✅ Récupérer tous les articles
    public List<Article> getAllArticles() {
        return articleRepository.findAll();
    }

    // ✅ Recherche un article par son nom
    public Optional<Article> getArticleByNom(String nom) {
        return articleRepository.findByNom(nom);
    }

    // ✅ Recherche des articles contenant un mot-clé
    public List<Article> searchArticles(String query) {
        return articleRepository.findByNomContainingIgnoreCase(query);
    }
}
